File: scripts/rename_admin_routes.py
# -*- coding: utf-8 -*-
import os

ROOT = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
SKIP_SUFFIX = "routes/web.php"


def process_file(path: str) -> bool:
    with open(path, "r", encoding="utf-8") as f:
        c = f.read()
    orig = c
    c = c.replace("admin/organization", "back-office")
    c = c.replace("admin/system/", "admin/")
    c = c.replace("url('admin/system')", "url('admin')")
    c = c.replace('url("admin/system")', 'url("admin")')
    if c != orig:
        with open(path, "w", encoding="utf-8", newline="\n") as f:
            f.write(c)
        return True
    return False


def main() -> None:
    for sub in ("app", "views"):
        root = os.path.join(ROOT, sub)
        for dirpath, _, files in os.walk(root):
            for name in files:
                if not name.endswith(".php"):
                    continue
                rel = os.path.relpath(os.path.join(dirpath, name), ROOT).replace("\\", "/")
                if rel == SKIP_SUFFIX or rel.endswith("/" + SKIP_SUFFIX):
                    continue
                p = os.path.join(dirpath, name)
                if process_file(p):
                    print(p)


if __name__ == "__main__":
    main()
    print("OK")

File: scripts/rename-admin-routes.php
<?php
declare(strict_types=1);

$roots = [__DIR__ . '/../app', __DIR__ . '/../views'];
$skip = ['routes/web.php'];

foreach ($roots as $root) {
    $it = new RecursiveIteratorIterator(new RecursiveDirectoryIterator($root));
    foreach ($it as $file) {
        if (!$file->isFile() || $file->getExtension() !== 'php') {
            continue;
        }
        $path = $file->getPathname();
        foreach ($skip as $s) {
            if (str_ends_with(str_replace('\\', '/', $path), $s)) {
                continue 2;
            }
        }
        $c = file_get_contents($path);
        $orig = $c;
        $c = str_replace('admin/organization', 'back-office', $c);
        $c = str_replace('admin/system/', 'admin/', $c);
        $c = str_replace("url('admin/system')", "url('admin')", $c);
        $c = str_replace('url("admin/system")', 'url("admin")', $c);
        if ($c !== $orig) {
            file_put_contents($path, $c);
            echo $path, "\n";
        }
    }
}
echo "OK\n";
